### PR TITLE
Implement client function on rust throughput example

### PR DESCRIPTION
**Pull request by goforlee**
What the function does is the same as do_client() function in throughput.c except printing out the very last result of the packet communication.

### CODE REVIEW

File: tsn/src/bin/throughput.rs [hunk 1]
```diff
@@ -264,6 +264,11 @@ fn statistics_thread() {
 }
 
 fn do_client(sock: &mut i32, iface: String, size: i32, target: String, time: i32) {
+    let mut pkt: Vec<u8> = vec![0; size as usize];
+    let ethernet_size = mem::size_of::<Ethernet>();
+    let pkt_info_size = mem::size_of::<PktInfo>();
+    let recv_packet_size = ethernet_size + pkt_info_size;
+
```
Comment (tribela) [inline tsn/src/bin/throughput.rs:271]: Good

File: tsn/src/bin/throughput.rs [hunk 2]
```diff
@@ -330,26 +335,21 @@ fn do_client(sock: &mut i32, iface: String, size: i32, target: String, time: i32
...
-    let mut pkt_info_bytes = bincode::serialize(&pkt_info).unwrap();
-    start_pkt.append(&mut pkt_info_bytes);
-
+    make_send_pkt(&mut pkt, &ethernet, &pkt_info, size as usize);
     let mut is_successful: bool = false;
 
     while !is_successful {
-        send_perf(sock, &mut start_pkt, size as usize);
+        send_perf(sock, &mut pkt, recv_packet_size);
         is_successful = recv_perf(
```
Comment (tribela) [inline tsn/src/bin/throughput.rs:348]: Could be better if recv_perf returns `Result`

Comment (tribela) [reply tsn/src/bin/throughput.rs:348]: Also, loop and break if matched(Result) is considered

File: tsn/src/bin/throughput.rs [hunk 3]
```diff
@@ -421,15 +420,16 @@ fn send_perf(sock: &i32, pkt: &mut Vec<u8>, size: usize) {
...
-fn make_send_pkt(ethernet_bytes: &Vec<u8>, pkt_info: &PktInfo) -> Vec<u8> {
-    let mut pkt: Vec<u8> = ethernet_bytes.clone();
-    println!("pkt = {:0x?}", pkt);
-    let mut pkt_info_bytes = bincode::serialize(pkt_info).unwrap();
-    println!("pkt_info_bytes = {:0x?}", pkt_info_bytes);
-    pkt.append(&mut pkt_info_bytes);
-    println!("pkt = {:0x?}", pkt);
-
-    pkt
+fn make_send_pkt(pkt: &mut Vec<u8>, ethernet: &Ethernet, pkt_info: &PktInfo, size: usize) {
```
Comment (tribela) [inline tsn/src/bin/throughput.rs:423]: Function name is hard to understand.
Is it for make a packet to send?

File: tsn/src/bin/throughput.rs [hunk 4]
```diff
@@ -421,15 +420,16 @@ fn send_perf(sock: &i32, pkt: &mut Vec<u8>, size: usize) {
...
+    let ethernet_bytes = bincode::serialize(&ethernet).unwrap();
+    let pkt_info_bytes = bincode::serialize(pkt_info).unwrap();
+    let ethernet_size = ethernet_bytes.len();
+    let pkt_info_size = pkt_info_bytes.len();
+
+    let new_len = ethernet_size + pkt_info_size;
+    pkt.splice(..ethernet_size, ethernet_bytes);
+    pkt.splice(ethernet_size..new_len, pkt_info_bytes);
+    pkt.truncate(size);
```
Comment (tribela) [inline tsn/src/bin/throughput.rs:432]: Consider `split_at` method. it is cleaner way to do this.
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_at

**Review by tribela — COMMENTED** [review]
Seems fine. But there are few lines that could be changed.

File: tsn/src/bin/throughput.rs (hunk 3 above)
Comment (goforlee) [reply tsn/src/bin/throughput.rs:416]: Yes, this function gets a packet ready to be sent

File: tsn/src/bin/throughput.rs (hunk 4 above)
Comment (goforlee) [reply tsn/src/bin/throughput.rs]: I used `split_at_mut` instead if that's okay

File: tsn/src/bin/throughput.rs (hunk 3 above)
Comment (tribela) [reply tsn/src/bin/throughput.rs]: "make" and "send" both are verb it is weird name to use as a function name

Comment (goforlee) [reply tsn/src/bin/throughput.rs]: How about prep_pkt?

**Pull request merged**